presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -178,7 +181,9 @@ public ExtendedHiveMetastore getMetastore()
     @Override
     protected org.apache.iceberg.Table getIcebergTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        return getHiveIcebergTable(metastore, hdfsEnvironment, session, schemaTableName);
+        return icebergTables.computeIfAbsent(
```

Comment:
To prevent code duplication, I think it would be better to keep these methods the same. Then implement the caching in the `AbstractIcebergMetadata` class.

You may consider renaming the abstract method to `getRawIcebergTable`, `getDirectIcebergTable` or something like that to prevent having to refactor all of the call sites.